app/assets/javascripts/models/beta/iteration.js

```diff
@@ -156,52 +196,54 @@ const pastAverageVelocity = (pastIterations) => {
 }
 
 const addToSprintFromBacklog = (sprints, project, story, velocity) => {
-  const sprintIndex = sprints.length && sprints.length - 1;
-  const hasSpace = canTakeStory(sprints[sprintIndex], Story.getPoints(story));
-
-  if (hasSpace) {
-    return addStoryToSprint(project, sprints, sprintIndex, story, velocity);
-  }
-
   const iterationNumber = getNextIterationNumber(sprints);
 
-  sprints[sprintIndex + 1] = createSprint(
+  return [...sprints, createSprint(
     iterationNumber,
     getDateForIterationNumber(iterationNumber, project),
-    undefined,
-    velocity
-  );
-
-  return addStoryToSprint(project, sprints, sprintIndex + 1, story, velocity);
+    false,
+    velocity,
+    [story],
+    Story.getPoints(story)
+  )];
 };
 
 export const groupBySprints = (stories = [], project, initialSprintNumber = 1, pastIterations) => {
   const velocity = sprintVelocity(project, pastIterations)
 
   return stories.reduce((sprints, story) => {
     const firstSprintIndex = 0;
-    const isFromSprintInProgress = !Story.isUnstarted(story);
-
-    sprints = handleSprintsOverflow(
-      project,
-      sprints,
-      Story.getPoints(story),
-      initialSprintNumber,
-      velocity
-    );
+    const isFromSprintInProgress = Story.isInProgress(story);
+
+    if (causesOverflow(story, velocity) && !isFromSprintInProgress) {
+      return addSprintWithOverflow(project, sprints, story, velocity, initialSprintNumber);
+    }
 
     if (sprints.length === 0) {
-      createFirstSprint(sprints, project, initialSprintNumber, velocity);
+      sprints = [createFirstSprint(project, initialSprintNumber, velocity)];
     }
```

Comment:
Correct me if I'm wrong but, this is only truthy in the first iteration, right?
Because the initial value is `[]` (`stories.reduce((sprints, story) => {...}, /*initialValue is */ []`).
The other possibility is if the methods `addSprintWithOverflow`, `addStoryToSprint` or `addToSprintFromBacklog` doesn't add any value in the array. It can happen? 
If not, you could remove this `if` and set `[createFirstSprint(project, initialSprintNumber, velocity)]` as the initial value:
```js
export const groupBySprints = (stories = [], project, initialSprintNumber = 1, pastIterations) => {
  const velocity = sprintVelocity(project, pastIterations)

  return stories.reduce((sprints, story) => {
    const firstSprintIndex = 0;
    const isFromSprintInProgress = Story.isInProgress(story);

    if (causesOverflow(story, velocity) && !isFromSprintInProgress) {
      return addSprintWithOverflow(project, sprints, story, velocity, initialSprintNumber);
    }

    if (isFromSprintInProgress) {
      return addStoryToSprint(project, sprints, firstSprintIndex, story, velocity);
    }

    if (canEnterLastAddedSprint(sprints, story, velocity)) {
      return addStoryToSprint(project, sprints, sprints.length - 1, story, velocity);
    }

    return addToSprintFromBacklog(sprints, project, story, velocity);
  }, [createFirstSprint(project, initialSprintNumber, velocity)]);
};
```

Reply:
I think than this is not possible, because how you said when we have no one stories, if we use `stories.reduce(..., [createFirstSprint(project, initialSprintNumber, velocity)]);` will return one position right? And in some cases we have no sprint.

But, I'm not sure about that. What you think @talyssonoc ?

Reply:
That's right, if the array to be reduced is empty it's gonna return the initial value, which won't work for our case.